### PR TITLE
Adding new test case for Azure cred change with PXB upgrade

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:

- [x] Added a new test case for Azure cred change upgrade scenario - `PXBackupUpgradeWithAzureCredChange`

**Which issue(s) this PR fixes** (optional)
Closes #PB-7698

**Special notes for your reviewer**:
[Jenkins BYOC run - upgraded from 2.6.0 to 2.7.2](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/Cloud_BYOC/528)
[Jenkins BYOC run - upgraded from 2.7.0 to 2.7.2](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/Cloud_BYOC/527)

### CODE REVIEW

File: tests/backup/backup_upgrade_test.go
Comment (kshithijiyer-px) [inline tests/backup/backup_upgrade_test.go:1677]: If not needed can we remove if needed then please add a TODO comment/

File: tests/backup_helper.go
Comment (kshithijiyer-px) [inline tests/backup_helper.go:4468]: Why do we need this? Ideally we can pass it from the job right?

File: tests/backup_helper.go
Comment (kshithijiyer-px) [inline tests/backup_helper.go:114]: Let's update it with 24.3.0

File: tests/backup/backup_upgrade_test.go
Comment (kshithijiyer-px) [inline tests/backup/backup_upgrade_test.go:1635]: Can you use a loop here?

File: tests/backup/backup_upgrade_test.go
Comment (kshithijiyer-px) [inline tests/backup/backup_upgrade_test.go:1574]: Quarter is wrong

**Review by snigdha-px — APPROVED** [review]
LGTM
Add a pass log

File: tests/backup/backup_upgrade_test.go
Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go]: This was commented out just for testing purpose because I just had one source cluster

Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go]: The comment will be removed before merging

File: tests/backup_helper.go
Comment (mkoppal-px) [reply tests/backup_helper.go:4468]: Added just for testing purpose because the BYOC pipeline did not have this parameter configured

File: tests/backup/backup_upgrade_test.go
Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go:1636]: Fixed it

File: tests/backup_helper.go
Comment (mkoppal-px) [reply tests/backup_helper.go:114]: We always used dev builds of stork. Never use GA version for testing as the latest version

File: tests/backup/backup_upgrade_test.go
Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go]: Fixed it

File: tests/backup/backup_upgrade_test.go
Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go]: Fixed it

File: tests/backup_helper.go
Comment (kshithijiyer-px) [reply tests/backup_helper.go:114]: okay you have a point. Ignore it.

File: tests/backup/backup_upgrade_test.go
Comment (kshithijiyer-px) [inline tests/backup/backup_upgrade_test.go:1698]: I would be good if we can move this into a function.

**Review by kshithijiyer-px — APPROVED** [review]
LGTM let's see if you can move it to a helper rest looks good to me.

File: tests/backup/backup_upgrade_test.go
Comment (ak-px) [inline tests/backup/backup_upgrade_test.go:1894]: better to make use of WaitForClusterDeletion here.

Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go]: Done

File: tests/backup/backup_upgrade_test.go
Comment (mkoppal-px) [reply tests/backup/backup_upgrade_test.go:1697]: I thought about that, but it is as good as just calling UpdateCloudCredential inside another helper function.

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

**Pull request merged**
